compiler/noirc_evaluator/src/ssa/interpreter/mod.rs

```diff
@@ -241,11 +238,24 @@ impl<'ssa, W: Write> Interpreter<'ssa, W> {
         Ok(())
     }
 
-    pub(crate) fn call_function(
+    /// Interpret an entry point, assuming the globals have already been interpreted.
+    ///
+    /// This resets any previous call stack and step counter.
+    pub(crate) fn interpret_function(
         &mut self,
         function_id: FunctionId,
-        mut arguments: Vec<Value>,
+        arguments: Vec<Value>,
     ) -> IResults {
+        self.step_counter = 0;
+        self.call_stack.truncate(1);
+        self.call_function(function_id, arguments)
```

Comment:
I'm wondering whether we need an assertion on here similar to how we do for `interpret_globals` to avoid the interpreter from being changed to call this internally rather than `call_function`.

Reply:
Not sure we can do this without requiring a 2 step process for what would now be an `interpret_function` call so 🤷